cpp/inc/bond/core/nullable.h

```diff
@@ -435,7 +427,7 @@ class nullable<T, Allocator, false>
     {
         if (_value)
         {
-            destroy(_alloc);
+            delete_value();
             _value = 0;
```

Comment:
Even those smart pointer types must satisfy the nullable pointer concept. Ideally, `nullptr` would be used here if C++03 compatibility was not required, but I'll use default construction which is equally readable.